catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformer/impl/WorkspaceTransformerImpl.java

```diff
@@ -128,12 +128,18 @@ private void addAttributeValue(Metacard metacard, Map.Entry<String, Object> entr
         entry.getKey(),
         value);
 
-    if (value instanceof Serializable) {
-      metacard.setAttribute(new AttributeImpl(entry.getKey(), (Serializable) value));
-    } else if (value instanceof List) {
-      metacard.setAttribute(new AttributeImpl(entry.getKey(), (List<Serializable>) value));
-    } else {
-      LOGGER.debug("The value was expected to be Serializable or a list of Serializable items.");
+    try {
+      if (value instanceof Serializable) {
+        metacard.setAttribute(new AttributeImpl(entry.getKey(), (Serializable) value));
+      } else if (value instanceof List) {
+        metacard.setAttribute(new AttributeImpl(entry.getKey(), (List<Serializable>) value));
+      } else {
+        LOGGER.debug(
+            "The {} was expected to be Serializable or a list of Serializable items.", value);
+      }
+    } catch (ClassCastException ex) {
```

Comment:
:question: I don't see how a `ClassCastException` would occur in this code; everything is type checked before casting. Is an implementation of something called in here like `setAttribute` throwing it?

Reply:
It is also not a good practice to catch `ClassCastException`

Reply:
It could happen because the contents of the list might not be `Serializable`.

Reply:
Then the code should check and protect itself and not just react to  a `ClassCastException`

Reply:
yeah, and it does... and it is not handled there.  This is adding some extra safety instead of relying on SetAttribute to handle it cleanly